test/toolkit/test_toolkit.go

```diff
@@ -0,0 +1,293 @@
+	pc := *p // Shallow clone
+	pc.expectError = true
+	pc.expectIncludes = items
+	return &pc
+}
+
+func (p *Params) ExpectIncludes(items ...string) *Params {
+	pc := *p // Shallow clone
+	pc.expectIncludes = items
+	return &pc
+}
+
+func (p *Params) ExpectEqual(item string) *Params {
+	if item == "" {
+		panic("Checking against empty string not supported in ExpectEqual")
+	}
+	pc := *p // Shallow clone
+	pc.expectOK = true
+	pc.expectEqual = item
+	return &pc
+}
+
+func (p *Params) ExpectEmptyJsonArray() *Params {
+	pc := *p // Shallow clone
+	pc.exectJsonArrayPropertyName = "NA"
+	pc.expectJsonArrayValues = []string{}
+	return &pc
+}
+func (p *Params) ExpectJsonArray(propertyName string, items ...string) *Params {
+	pc := *p // Shallow clone
+	pc.exectJsonArrayPropertyName = propertyName
+	pc.expectJsonArrayValues = items
+	return &pc
+}
+
+func (p *Params) Describe(title string) *Params {
+	p.T.Log(title)
+	return p
+}
+
+func toGoldenFileName(name string) string {
+	goldenBaseName := strings.Replace(name, "/", "_", -1)
+	return goldenBaseName + ".golden"
+}
+
+func (p *Params) filterForGold(content string) string {
+	for _, x := range goldPolishers {
+		content = x(content)
+	}
+	return content
+}
+
+func (p *Params) WithParams(newP Params) Params {
+	pc := *p // Shallow clone
+	if newP.Config != "" {
+		pc.Config = newP.Config
+	}
+	if newP.Engine != "" {
+		pc.Engine = newP.Engine
+	}
+	if newP.App != "" {
+		pc.App = newP.App
+	}
+	if newP.Verbose != "" {
+		pc.Verbose = newP.Verbose
+	}
+	if newP.Traffic != "" {
+		pc.Traffic = newP.Traffic
+	}
+	if newP.Ttl != "" {
+		pc.Ttl = newP.Ttl
+	}
+	if newP.Headers != "" {
+		pc.Headers = newP.Headers
+	}
+	if newP.NoData != "" {
+		pc.NoData = newP.NoData
+	}
+	return pc
+}
+
+func (p *Params) Run(command ...string) []byte {
+	var output []byte
+	name := strings.Join(command, " ")
+	p.T.Run(name, func(t *testing.T) {
+		var args []string
+		if p.App != "" {
+			args = append(args, "--app", p.App)
+		}
+		if p.Engine != "" {
+			args = append(args, "--engine", p.Engine)
+		}
+		if p.Config != "" {
+			args = append(args, "--config", p.Config)
+		}
+		if p.Verbose == "true" {
+			args = append(args, "--verbose")
+		}
+		if p.Headers != "" {
+			args = append(args, "--headers", "\""+p.Headers+"\"")
+		}
+		if p.NoData == "true" {
+			args = append(args, "--no-data")
+		}
+		if p.Traffic == "true" {
+			args = append(args, "--traffic")
+		}
+		if p.Ttl != "" {
+			args = append(args, "--ttl", p.Ttl)
+		}
+		args = append(args, command...)
+
+		createdApp := createsApp(args)
+
+		if createdApp != "" {
+			createdApps = append(createdApps, createdApp)
+		}
+
+		cmd := exec.Command(binaryPath, args...)
+
+		goldenName := toGoldenFileName(t.Name())
+
+		t.Log("\u001b[35m Executing command:" + strings.Join(cmd.Args, " ") + "\u001b[0m")
+		var err error
+		output, err = cmd.CombinedOutput()
+
+		actual := string(output)
+		t.Log("\u001b[35m Output:\n" + actual)
```

Comment:
Coloring shoud be disabled after logging:
`t.Log("\u001b[35m Output:\n" + actual + "\u001b[0m")`